docs/apache-airflow/concepts/dynamic-task-mapping.rst

```diff
@@ -0,0 +1,256 @@
+    # This results in add function being expanded to
+    # add(x=1,y=10)
+    # add(x=2,y=10)
+    # add(x=3,y=10)
+
+This would result in values of 11, 12, 13.
+
+This is also useful for passing things such as connection IDs, database table names, or bucket names to tasks.
+
+Mapping over multiple parameters
+================================
+
+As well as a single parameter it is possible to pass multiple parameters to expand. This will have the effect of creating a "cross product" effect, calling the mapped task with each combination of parameters.
+
+For example:
+
+.. code-block:: python
+
+    @task
+    def add(x: int, y: int):
+        return x + y
+
+
+    added_values = add.expand(x=[2, 4, 8], y=[5, 10])
+    # This results in the add function being called with
+    # add(x=2, y=5)
+    # add(x=2, y=10)
+    # add(x=4, y=5)
+    # add(x=4, y=10)
+    # add(x=8, y=5)
+    # add(x=8, y=10)
+
+This would result in the add task being called 6 times. Please note however that the order of expansion is not guaranteed.
+
+It is not possible to achieve an effect similar to Python's zip function with mapped arguments.
+
+Task-generated Mapping
+======================
+
+Up until now the examples we've shown could all be achieved with a ``for`` loop in the DAG file, but the real power of dynamic task mapping comes from being able to have a task generate the list to iterate over. For example:
+
+.. code-block:: python
+
+    @task
+    def make_list():
+        # This can also be from an API call, checking a database, -- almost anything you like, as long as the
+        # resulting list/dictionary can be stored in the current XCom backend.
+        return [1, 2, {"a": "b"}, "str"]
+
+
+    @task
+    def consumer(arg):
+        print(repr(arg))
+
+
+    with DAG(dag_id="dynamic-map", start_date=datetime(2022, 4, 2)) as dag:
+        consumer.expand(arg=make_list())
+
+The make_list runs as a normal task and must return a list or dict (see `What data types can be expanded?`_), and then the consumer task will be called four times, once with each value in the return of make_list.
+
+Mapping with non-TaskFlow operators
+===================================
+
+It is possible to use partial and expand with classic style operators as well. Some arguments are not mappable, such as task_id, queue, pool, and most other arguments to BaseOperator and have to be passed to ``partial()``
+
+
+.. code-block:: python
+
+    BashOperator.partial(task_id="bash", do_xcom_push=False).expand(
+        bash_command=["echo 1", "echo 2"]
+    )
+
+Mapping over result of classic operators
+----------------------------------------
+
+If you want to map over the result of a classic operator you will need to create an XComArg object manually:
+
+.. code-block:: python
+
+    from airflow import XComArg
+
+    task = MyOperator(task_id="source")
+
+    downstream = MyOperator2.partial(task_id="consumer").expand(input=XComArg(task))
```

Comment:
this example is a bit confusing.   not clear why you have to use an xcom arg or what an xcom arg is.  could the task -> xcomarg conversion not be handled in `expand`?

Reply:
It likely could, yes. Right now I'm just going with documenting what the current behaviour is.